src/components/ColumnSettingsList.ts

```diff
@@ -0,0 +1,280 @@
+        end: ['Space', 'Enter'],
+      },
+    })
+  );
+
+  const renderItem = (item, index) => {
+    return h(SortableItem, { id: item.id, name: item.name, index: getPosition(item.id) }, [
+      h(DragHandle),
+      h(Fragment, [
+        h(
+          TooltipTrigger,
+          {
+            // Since attribute names don't contain spaces, word-break: break-all is necessary to
+            // wrap the attribute name instead of truncating it when the tooltip reaches its
+            // max width of 400px.
+            content: span({ style: { wordBreak: 'break-all' } }, [item.name]),
+          },
+          [
+            label(
+              {
+                style: {
+                  lineHeight: `${rowHeight}px`,
+                  ...Style.noWrapEllipsis,
+                },
+              },
+              [
+                h(Checkbox, {
+                  'aria-label': `Show "${item.name}" in table`,
+                  checked: item.visible,
+                  onChange: () => {
+                    if (!_.isUndefined(index)) toggleVisibility(index);
+                  },
+                }),
+                ' ',
+                item.name,
+              ]
+            ),
+          ]
+        ),
+      ]),
+    ]);
+  };
+
+  // Utilities and support related to screen reader text.
+  const getPosition = (id) => _.findIndex({ id })(items) + 1;
+  const getName = (id) => {
+    const item = _.filter({ id })(items) as unknown;
+    return item ? item[0].name : 'unknown';
+  };
+  const announcements = <Announcements>{
+    onDragStart({ active }) {
+      return `Picked up "${getName(active.id)}". Column "${getName(active.id)}" is in position ${getPosition(
+        active.id
+      )} of ${items.length}.`;
+    },
+    onDragOver({ active, over }) {
+      if (over) {
+        return `"${getName(active.id)}" was moved into position ${getPosition(over.id)} of ${items.length}.`;
+      }
+    },
+    onDragEnd({ active, over }) {
+      if (over) {
+        return `"${getName(active.id)}" was dropped at position ${getPosition(over.id)} of ${items.length}.`;
+      }
+    },
+    onDragCancel({ active }) {
+      return `Dragging was cancelled. "${getName(active.id)}" was dropped.`;
+    },
+  };
+  const screenReaderInstructions = <ScreenReaderInstructions>{
+    draggable: `
+    To begin a drag, press the space bar or Enter.
+    While dragging, use the up and down arrow keys to move the item.
+    Press space or Enter again to drop the item in its new position, or press the letter q to cancel.
+    `,
+  };
+
+  const onDragEnd = ({ active, over }) => dragEndNotifier({ active, over, items, onChange });
+
+  return h(AutoSizer, { disableHeight: true }, [
+    ({ width }) => {
+      return h(DndContext, { sensors, onDragEnd, accessibility: { announcements, screenReaderInstructions } }, [
+        h(SortableContext, { items }, [
+          // @ts-ignore
```

Comment:
Generally, `ts-expect-error` would be preferable over `ts-ignore` so that we get notified to remove it if we resolve whatever causes TS to complain.

Reply:
Done!